geode-protobuf/src/main/java/org/apache/geode/protocol/protobuf/operations/GetAvailableServersOperationHandler.java

```diff
@@ -50,51 +37,23 @@
   @Override
   public Result<ServerAPI.GetAvailableServersResponse> process(
       SerializationService serializationService, ServerAPI.GetAvailableServersRequest request,
-      Cache cache) {
-
-    InternalDistributedSystem distributedSystem =
-        (InternalDistributedSystem) cache.getDistributedSystem();
-    Properties properties = distributedSystem.getProperties();
-    String locatorsString = properties.getProperty(ConfigurationProperties.LOCATORS);
+      MessageExecutionContext executionContext) throws InvalidExecutionContextException {
 
-    HashSet<DistributionLocatorId> locators = new HashSet();
-    StringTokenizer stringTokenizer = new StringTokenizer(locatorsString, ",");
-    while (stringTokenizer.hasMoreTokens()) {
-      String locator = stringTokenizer.nextToken();
-      if (StringUtils.isNotEmpty(locator)) {
-        locators.add(new DistributionLocatorId(locator));
-      }
+    InternalLocator locator = executionContext.getLocator();
+    ArrayList serversFromSnapshot =
```

Comment:
I don't like that this has to go so far in to get a list of servers. I'd like to see the DUnit test (or an integration test using a real Locator) come back just so we can have some guarantee that `InternalLocator` doesn't change on us.